src/index.js

```diff
@@ -5,1981 +5,235 @@
+  unregisterTransactionRelay: function (relayer) { return dispatch(unregisterTransactionRelay(relayer)); },
+  excludeFromTransactionRelay: function (method) { return dispatch(excludeFromTransactionRelay(method)); },
+  includeInTransactionRelay: function (method) { return dispatch(includeInTransactionRelay(method)); },
 
   /******************************
    * Ethereum JSON-RPC bindings *
    ******************************/
 
-  raw: function (command, params, callback) {
-    var transportRequirements = "ANY";
-    if (!callback) transportRequirements = "SYNC";
-    return this.submitRequestToBlockchain(this.marshal(command, params, null), transportRequirements, callback);
-  },
-
-  eth: function (command, params, callback) {
-    return this.raw("eth_" + command, params, callback);
-  },
-
-  net: function (command, params, callback) {
-    return this.raw("net_" + command, params, callback);
-  },
-
-  web3: function (command, params, callback) {
-    return this.raw("web3_" + command, params, callback);
-  },
-
-  shh: function (command, params, callback) {
-    return this.raw("shh_" + command, params, callback);
-  },
-
-  miner: function (command, params, callback) {
-    return this.raw("miner_" + command, params, callback);
-  },
-
-  admin: function (command, params, callback) {
-    return this.raw("admin_" + command, params, callback);
-  },
-
-  personal: function (command, params, callback) {
-    return this.raw("personal_" + command, params, callback);
-  },
-
-  txpool: function (command, params, callback) {
-    return this.raw("txpool_" + command, params, callback);
-  },
-
-  // ****
-  // web3_*
-  // ****
-
-  clientVersion: function (callback) {
-    return this.web3("clientVersion", null, callback);
-  },
-
-  // TODO: make this take a callback like everything else
-  sha3: function (data, isHex) {
-    if (isHex) data = abi.decode_hex(data);
-    return abi.prefix_hex(keccak_256(data));
-  },
-
-  // ****
-  // net_*
-  // ****
-
-  listening: function (callback) {
-    return this.net("listening", null, callback);
-  },
-
-  peerCount: function (callback) {
-    return this.net("peerCount", null, callback);
-  },
-
-  version: function (callback) {
-    return this.net("version", null, callback);
-  },
-
-  netVersion: function (callback) {
-    return this.version(callback);
-  },
-
-  // ****
-  // eth_*
-  // ****
-
-  accounts: function (callback) {
-    return this.eth("accounts", null, callback);
-  },
-
-  blockNumber: function (callback) {
-    return this.eth("blockNumber", null, callback);
-  },
-
+  raw: function (cmd, params, callback) { return dispatch(raw(cmd, params, callback)); },
+
+  eth: bindDispatch(dispatch, eth),
+  net: bindDispatch(dispatch, net_),
+  web3: bindDispatch(dispatch, web3),
+  txpool: bindDispatch(dispatch, txpool),
+  shh: bindDispatch(dispatch, shh),
+  miner: bindDispatch(dispatch, miner),
+  personal: bindDispatch(dispatch, personal),
+
+  // web3_
+  sha3: sha3,
```

Comment:
It's because `sha3` has optimized by doing the calculation locally, so I see no real use-case for the actual `web3_sha3` RPC.